hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java

```diff
@@ -518,23 +542,29 @@ public Optional<ExceptionalHaltReason> tryTrackingSelfDestructBeneficiary(
         }
         final AccountID accountID = AccountID.newBuilder().accountNum(number).build();
         final var account = nativeOperations.getAccount(accountID);
-        if (account == null) {
-            final var token = nativeOperations.getToken(number);
-            if (token != null) {
-                // If the token is deleted or expired, the system contract executed by the redirect
-                // bytecode will fail with a more meaningful error message, so don't check that here
-                return new TokenEvmAccount(address, this);
+        if (account != null) {
+            if (account.deleted() || account.expiredAndPendingRemoval() || isNotPriority(address, account)) {
+                return null;
+            }
+            if (account.smartContract()) {
+                return new ProxyEvmContract(account.accountId(), this);
+            } else {
+                return new ProxyEvmAccount(account.accountId(), this);
             }
-            return null;
         }
-        if (account.deleted() || account.expiredAndPendingRemoval() || isNotPriority(address, account)) {
-            return null;
+        final var token = nativeOperations.getToken(number);
+        if (token != null) {
+            // If the token is deleted or expired, the system contract executed by the redirect
+            // bytecode will fail with a more meaningful error message, so don't check that here
+            return new TokenEvmAccount(address, this);
         }
-        if (account.smartContract()) {
-            return new ProxyEvmContract(account.accountId(), this);
-        } else {
-            return new ProxyEvmAccount(account.accountId(), this);
+        final var schedule = nativeOperations.getSchedule(number);
+        if (schedule != null) {
+            // If the schedule is deleted or expired, the system contract executed by the redirect
```

Comment:
(Now that I've seen this same comment half-a-dozen times I took a look at the entire file as it is without diffs and this class makes me sad.  Lots of nested ifs, lots of sequential ifs.  Lots of that is error checking code of course, but it also obscures the actual work being done as well as making it annoying to read the code.  Error checking code for business rules as well as programmer errors - both of which should be _ubquitous_ in most production systems - are a real pain point in all computer languages - tedious to write, annoying to verify, obscures useful logic, etc. - but apparently this has never been important to language designers or academic programming language researchers ...)

Reply:
This class could use some refactoring, I agree.